evennia/contrib/utils/database_backup/database_backup.py

```diff
@@ -0,0 +1,184 @@
+from django.conf import settings
+
+from evennia.comms.models import ChannelDB
+from evennia import CmdSet, DefaultScript
+from evennia.commands.default.muxcommand import MuxCommand
+from evennia.utils.create import create_script
+from evennia.utils import logger, search
+
+import datetime
+import os
+import shutil
+import subprocess
+
+_MUDINFO_CHANNEL = None
+BACKUP_FOLDER = "server/backups"
+DATETIME_FORMAT_STR = "%Y-%m-%d.%H_%M_%S"
+DEFAULT_INTERVAL = 86400
+
+
+class DatabaseBackupScript(DefaultScript):
+    """
+    The global script to backup the server on a schedule.
+
+    It will be automatically created the first time the `backup` command is used.
+    """
+
+    def at_script_creation(self):
+        super().at_script_creation()
+        self.key = "db_backup_script"
+        self.desc = "Database backups"
+        self.persistent = True
+
+    def log(self, message):
+        global _MUDINFO_CHANNEL
+        if not _MUDINFO_CHANNEL and settings.CHANNEL_MUDINFO:
+            channels = search.search_channel(settings.CHANNEL_MUDINFO["key"])
+            if channels:
+                _MUDINFO_CHANNEL = channels[0]
+
+        if _MUDINFO_CHANNEL:
+            _MUDINFO_CHANNEL.msg(message)
+        logger.log_sec(message)
+
+    def backup_postgres(self, db_name, db_user, output_file_path):
+        """
+        Run `pg_dump` on the postgreSQL database and save the output.
+        """
+
+        output_file_path += ".sql"
+        subprocess.run(
+            ["pg_dump", "-U", db_user, "-F", "p", db_name],
+            stdout=open(output_file_path, "w"),
+            check=True,
+        )
+        self.log(f"|wpostgresql db backed up in: {BACKUP_FOLDER}|n")
+
+    def backup_sqlite3(self, db_name, output_file_path):
+        """
+        Copy the sqlite3 db.
+        """
+
+        output_file_path += ".db3"
+        os.makedirs(os.path.dirname(output_file_path), exist_ok=True)
+        shutil.copy(db_name, output_file_path)
+        self.log(f"|wsqlite3 db backed up in: {BACKUP_FOLDER}|n")
```

Comment:
How safe is this operation? The sqlite3 file is just a binary and `shutil.copy` doesn't know anything about what a database is. What happens if this is being written to (by an active game) while this copy operation is performed?